src/StringHelper.php

```diff
@@ -473,4 +473,18 @@ public static function base64UrlDecode(string $input): string
     {
         return base64_decode(strtr($input, '-_', '+/'));
     }
+
+    /**
+     * Split a string to array with non-empty lines.
+     * Whitespace from the beginning and end of a each line will be stripped.
+     *
+     * @param string $string The input string.
+     *
+     * @return array
+     */
+    public static function split(string $string): array
+    {
+        $string = preg_replace('(^\s*|\s*$)', '', $string);
+        return preg_split('~\s*\R\s*~', $string, -1, PREG_SPLIT_NO_EMPTY);
```

Comment:
```suggestion
        return preg_split('~\s*\R\s*~', trim($string), -1, PREG_SPLIT_NO_EMPTY);
```

Reply:
`trim` not equal preg regular exceptions `\s`.

`trim`: characters 0, 9, 10, 11, 13, 32 (https://www.php.net/manual/en/function.trim.php)

`\s`:  characters 9, 10, 12, 13, 32 + specific: "However, if locale-specific matching is happening, characters with code points in the range 128-255 may also be considered as whitespace characters, for instance, NBSP (A0)." (https://www.php.net/manual/en/regexp.reference.escape.php).

In some cases, this may lead to illogical behavior.

Reply:
Add a test for it.

Reply:
> Add a test for it.

Done. For string `StringHelper::split("\0\nA\nB")` return different result for `trim` and `preg_replace` with `\s`.